manifests/repo.pp

```diff
@@ -1,148 +1,239 @@
-# == Define: yum::repo
+# @summary Manage individual yum repo files in /etc/yum.repos.d
+#   This was used in favor of the yumrepo type, which cannot
+#   manage files in that directory.
 #
-# Manage individual yum repo files in /etc/yum.repos.d
+# @example Using the define.
+#   yum::repo { 'redhat-base':
+#     gpgcheck => true,
+#   }
 #
-# This was used in favor of the yumrepo type, which cannot
-# manage files in that directory.
+# @param ensure
+#   Set to "present" or "absent". When absent, removes the repository
+#   configuration file from the node modeled on the behavior of the
+#   File type's ensure parameter.
 #
-# example
-#  yum::repo { 'redhat-base':
-#    gpgcheck => '1',
-#  }
+# @param repo_file_mode
+#   Set the file mode of the repository configuration file.
+#
+# @param yum_repos_d_path
+#   Specify the path of the directory for yum repository files.
+#
+# @param gpgkey_local_path
```

Comment:
this does not seem to be used, also I can't find documentation for it

Reply:
ah this seems to be part of the old module where it would build a URL for you and download and install the gpg key for you. We want to remove all of that old functionality and just support whatever the man page says is supported. We can then document what steps are needed to download and install a key with `yum::rpm_gpg_key`

Reply:
With "Deprecate GPG key download in repo define" the old functionality is removed completely.
Have added `$rpm_gpg_keys` and `$rpm_gpg_keys_hiera_merge` for easy GPG key management.
Documentation is updated accordingly.